tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -391,18 +331,151 @@ stages:
+        artifactPipeline: ${{ variables.artifactPipeline }}
+        customSteps:
+
+        # Download test package
+        - task: DownloadPipelineArtifact@2
+          displayName: Download test package
+          inputs:
+            # It seems there's a bug and preferTriggeringPipeline is not respected.
+            # We force the behavior by explicitly specifying:
+            # - buildVersionToDownload: specific
+            # - buildId: <the id of the triggering build>
+            # preferTriggeringPipeline: true
+            source: specific
+            project: internal
+            pipeline: ${{ variables.artifactPipeline }}
+            buildVersionToDownload: specific
+            buildId: ${{ variables.artifactBuildId }}
+            artifact: pack
+            patterns: "**/${{ variables.escapedTestPackage }}-?.?.?-*.tgz"
+            path: $(Pipeline.Workspace)/client/pack
+            # allowPartiallySucceededBuilds: true # No effect as long as we have buildVersionToDownload: specific
+            # branchName: $(Build.SourceBranch)   # No effect as long as we have buildVersionToDownload: specific
+
+        # Install e2e test package
+        - task: Bash@3
+          displayName: Install Test Package
+          inputs:
+            targetType: 'inline'
+            workingDirectory: ${{ variables.testWorkspace }}
+            script: |
+              echo "Install ${{ variables.testPackage }}"
+              TEST_PACKAGE_FILE_PATTERN=${{ variables.escapedTestPackage }}-?.?.?-*.tgz
+              TEST_PACKAGE_PATH_PATTERN=$(Pipeline.Workspace)/client/pack/scoped/$TEST_PACKAGE_FILE_PATTERN
+
+              echo $TEST_PACKAGE_FILE_PATTERN
+              echo $TEST_PACKAGE_PATH_PATTERN
+
+              echo `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l`
+              echo `ls $TEST_PACKAGE_PATH_PATTERN`
+
+              if [[ `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l` -eq 1 ]]; then
+                TEST_PACKAGE_TGZ=`ls $TEST_PACKAGE_PATH_PATTERN`
+              else
+                ls -1 $TEST_PACKAGE_PATH_PATTERN
+                echo "##vso[task.logissue type=error]Test package '${{ variables.testPackage }}' not found, or there are more then one found"
+                exit -1
+              fi
+
+              npm install $TEST_PACKAGE_TGZ
+
+        - task: Bash@3
+          displayName: Prepare test package to run tests
+          inputs:
+            targetType: 'inline'
+            workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ variables.testPackage }}
+            script: |
+              cp ${{ variables.testWorkspace }}/.npmrc . ;
+              npm i ;
+
+        - task: Bash@3
+          displayName: Run execution time tests
+          inputs:
+            targetType: 'inline'
+            workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ variables.testPackage }}
+            script: |
+              npm run test:benchmark:report;
+
+        - task: Bash@3
```

Comment:
does this run if the previous task failed? ideally we want it to run, so that we still report memory tests even if execution time test command fails. I don't think we want `succeededOrFailed()` here either, since we don't want to run it if other deps failed.

I'd guess no because the azure-pipeline `Bash` task returns the result of the script it's running, which will end up being nonzero if there are test failures. But I'm not particularly familiar with bash, so maybe worth testing. I'm fine doing that in a follow-up if you want, since this issue is pretty minor if it exists.

One potential fix if it is an issue would be to combine both runs into a single bash script, which should give more flexibility over which commands to run than azure pipeline yaml has.

Reply:
Good point, I'll look a bit into that topic.

Reply:
Ended up putting both in a single bash step and expanding the script to better report failures for one or the other.